### ISSUE TITLE
[Feature request] Support for special letters

### ISSUE THREAD

**Issue by mavethee** [RPCS3/rpcs3#15552]
### Quick summary

~~This should be more reasonable than previous one, I hope.~~
I've noticed that RPCS3 displays some special letters during trophy notifications but not all of them and I'm not exactly sure whats the issue here so, request can be renamed if needed.

### Details

Basically so far, I've noticed only `ć` displays correctly but `ą, ę, ś, ź ż` not really, it's just a gap in trophy notification, it's probably bigger issue but least I can confirm is just polish.

During GTA V finale `Żyć lub zginąć w Los Santos` (`To Live or Die in Los Santos`) - missing `Ż`, while `ć` is diplayed.
![image](https://github.com/RPCS3/rpcs3/assets/47083527/b8f39c5a-0473-45d3-8bfc-feb30847dbef)

Heavy Rain - `Rzeźnik` (The Buther) trophy - missing `ź` here.
![image](https://github.com/RPCS3/rpcs3/assets/47083527/39555415-7945-4fed-82dc-589db6730f22)

Just for the note, in trophy manager it displays everything correctly:
![CleanShot 2024-05-04 at 19 27 31@2x](https://github.com/RPCS3/rpcs3/assets/47083527/79eb9f2e-9dc7-4451-b356-5218939223f4)

**Comment by mavethee** [comment]
It turns out to be macOS only issue.
See [FlexBy's vid](https://www.youtube.com/watch?v=ZDYyEISsC20&t=4s) from RPCN's test in All 4 One, `Connected` trophy displays fine.

![Image](https://github.com/user-attachments/assets/928fd1ea-20d6-48af-bf2c-55cbf89b4163)

But `Russian` is even worse with bunch of unreable squares - during trophy sync, text isn't even visible.
![Image](https://github.com/user-attachments/assets/5d669d43-96b2-40f0-94a3-73efeb245b1d)

**Comment by kd-11** [comment]
We use PS3 fonts + system fallback. I'm honestly not sure anyone ever checked what happens when running on MacOS and accessing a codepage outside the ones found in the default font. Rpcs3 works with raw ttf fonts, I'm guessing MacOs doesn't use those?

**Comment by mavethee** [comment]
> We use PS3 fonts + system fallback. I'm honestly not sure anyone ever checked what happens when running on MacOS and accessing a codepage outside the ones found in the default font. Rpcs3 works with raw ttf fonts, I'm guessing MacOS doesn't use those?

@kd-11 It does, you can install ttfs by dragging them to the Font Album (? kill me I don't know the English name of this macOS app) or via Terminal (at least guessing by StackOverflow posts, I've installed a few via that app).

Issue probably is Apple doesn't supply the same ones by default as Microsoft, ig?

**Comment by kd-11** [comment]
Yes, that's likely the case. We also need to check the font search paths are correct for macos.

**Comment by mavethee** [comment]
@kd-11 okay so not sure if i did my digging correct, but checked paths they seem to be correct at least i'm not seem to missing any required font 🤔

**Comment by mavethee** [comment]
Still seems to be an issue, now with RPCS3 being translated, it's gets worse.

**Comment by Megamouse** [comment]
You just need to find a font that contains these on macOs

**Comment by mavethee** [comment]
@Megamouse well, Sony's FW fonts does if I understood correctly what we're looking for.
All `LATIN2` in `/dev_flash/data/font` do, I assume overlay uses LATIN.TFF and system wide failback?
Sorry if i'm totally missing the point but I'm a bit dumb scrapping RPCS3's source code whats causing that 🧠 

<img width="2750" height="1722" alt="Image" src="https://github.com/user-attachments/assets/ed095982-7280-4945-9165-4cca48609d25" />

Working:

`ć, ł, ó`

Not working:

`ą, ę, ń, ś, ż`

**Comment by kd-11** [comment]
That's pretty accurate. We use PS3 fonts with system fallback to find the required glyphs.
